test/cpp/tensorexpr/tutorial.cpp

```diff
@@ -0,0 +1,426 @@
+    //   independent computations over the same domain) for its elements, as a
+    //   function of indices
+    //
+    // We use Function objects to represent this. Let's build one.
+    //
+    // First, we need to specify the domain, or dimensions in which the
+    // computation would be performed. Let's create a 64x32 domain:
+    std::vector<const Expr*> dims = {
+        new IntImm(64), new IntImm(32)}; // IntImm stands for Integer Immediate
+                                         // and represents an integer constant
+
+    // Next we need to create Function arguments. The arguments of a Function
+    // are Vars, and they play role of placeholders. The computation that the
+    // function would describe would use these arguments.
+    const Var* i = new Var("i", kInt);
+    const Var* j = new Var("j", kInt);
+    std::vector<const Var*> args = {i, j};
+
+    // Now we can define the function computations using these arguments. Let's
+    // create two computations, the first would add the arguments of the
+    // function, the second would multiply them.
+    Expr* func_body1 = new Mul(i, j);
+    Expr* func_body2 = new Add(i, j);
+
+    // Finally, we pass all these pieces together to Function constructor:
+    Function* func =
+        new Function({"X", "Y"}, dims, args, {func_body1, func_body2});
+    // Under the hood function constructor would create separate `Buf`
+    // expressions for each computation (which can be accessed via
+    // `func->func_var(idx)`) with the names specified by the first parameter of
+    // the constructor call. In our example two `Buf` variables will be created
+    // with names 'X' and 'Y', each of them would signify a domain of 64x32.
+
+    // We can now print out our function:
+    std::cout << "Tensor function: " << *func << std::endl;
+    // Prints:
+    // Tensor function: Function F(i[64], j[32]) {
+    //   X = i * j
+    //   Y = i + j
+    // }
+
+    // A Tensor refers to an individual computation defined by a Function. For
+    // instance, we could create a following tensor given the function above:
+    int output_idx = 0; // Used to index the computation
+    Tensor* X = new Tensor(func, output_idx);
+    std::cout << "Tensor computation: " << *X << std::endl;
+    // Prints: Tensor computation: Tensor X(i[64], j[32]) = i * j
+
+    // Similarly to how we provide a more convenient way of using handles for
+    // constructing Exprs, Tensors also have a more convenient API for
+    // construction. It is based on Compute functions, which take a name:
+    // dimensions, and a lambda specifying the computation body:
+    Tensor* Z = Compute(
+        "Z",
+        {{64, "i"}, {32, "j"}},
+        [](const VarHandle& i, const VarHandle& j) { return i / j; });
+    std::cout << "Tensor computation: " << *Z << std::endl;
+    // Prints: Tensor computation: Tensor Z(i[64], j[32]) = i / j
+
+    // Tensors might access other tensors and external placeholders in their
```

Comment:
While the term "placeholder" has been mentioned informally above, it hasn't been formally defined yet